tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -633,27 +633,35 @@ static int bits_logical(void)
 
 /* gpio_pin_get_config()
  */
+#ifdef CONFIG_GPIO_GET_CONFIG
 static int pin_get_config(void)
 {
 	gpio_flags_t flags_get = 0;
 	gpio_flags_t flags_set;
 	int rc;
 
+	/* GPIO_INT_MASK		bit 26-21
+	 * GPIO_OUTPUT_INIT_HIGH, LOW	bit 19, 18
+	 * GPIO_OUTPUT, GPIO_INPUT	bit 17, 16
+	 * GPIO_PULL_DOWN, GPIO_PULL_UP	bit 5, 4
+	 * GPIO_LINE_OPEN_DRAIN/SOURCE	bit 2
+	 * GPIO_SINGLE_ENDED/PUSH_PULL	bit 1
+	 * GPIO_ACTIVE_LOW/HIGH		bit 0
+	 */
+	gpio_flags_t flags_mask = GENMASK(26, 16) | GENMASK(5, 4) | GENMASK(2, 0);
+
 	flags_set = GPIO_OUTPUT_HIGH;
 	rc = gpio_pin_configure(dev, PIN_OUT, flags_set);
 	zassert_equal(rc, 0, "pin configure failed");
 
 	rc = gpio_pin_get_config(dev, PIN_OUT, &flags_get);
-	if (rc == -ENOTSUP) {
```

Comment:
This should probably be kept. The reason is that there could be multiple GPIO drivers and one of them might support this feature, while the other does not.

That's the main reason why I kept the `gpio_get_direction()` tests in a separate suite.

Reply:
Only one GPIO driver gpio_stm32_get_config() will return ENOTSUP. When the runtime pm is not defined, the function will return ENOTSUP, and the gpio_pin_configure will not be performed. To my understanding, at this time, the test case should be regarded as TC_FAIL. Do you agree?

Reply:
This part is modified in https://github.com/zephyrproject-rtos/zephyr/pull/48159/files. So in this PR, I will not touch this.